subprojects/core/src/main/java/org/gradle/api/internal/changedetection/state/ZipHasher.java

```diff
@@ -142,8 +267,35 @@ private ZipEntryRelativePath(ZipEntry zipEntry) {
         }
     }
 
-    private HashCode hashMalformedZip(RegularFileSnapshot zipFileSnapshot, Exception e) {
-        LOGGER.debug("Malformed archive '{}'. Falling back to full content hash instead of entry hashing.", zipFileSnapshot.getName(), e);
-        return zipFileSnapshot.getHash();
+    private static class MapEntry implements Factory<String[]>, Comparable<MapEntry> {
```

Comment:
We don't need this. There should be no duplicate key situation you mention elsewhere, as both `Properties` and `Attributes` implement `Map`. We also don't need `Factory<String[]>` anymore, as `ResourceEntryHasher` doesn't need it.

Reply:
Oops, yeah forgot to drop that `Factory`.

I'm really sure there's a another reason this was necessary, let me take a look.

Reply:
Nope, looks fine. Dropped.